js/PaymentRequest/index.js

```diff
@@ -258,9 +271,44 @@ export default class PaymentRequest {
         INTERNAL_SHIPPING_ADDRESS_CHANGE_EVENT,
         this._handleShippingAddressChange.bind(this)
       );
+
+      this._paymentMethodChangeSubscription = DeviceEventEmitter.addListener(
+          INTERNAL_PAYMENT_METHOD_CHANGE_EVENT,
+          this._handlePaymentMethodChange.bind(this)
+      );
     }
   }
 
+  _handlePaymentMethodChange(paymentMethod: PaymentMethod) {
+    this._paymentMethod = paymentMethod.paymentMethodType;
+
+    if(this._paymentMethod === 'PKPaymentMethodTypeCredit') {
```

Comment:
Follow-up: This should be moved up to the application layer and the library should be more generic.